pandas/io/formats/csvs.py

```diff
@@ -33,27 +34,26 @@ class CSVFormatter:
     def __init__(
         self,
         obj,
-        path_or_buf=None,
-        sep=",",
-        na_rep="",
-        float_format=None,
+        path_or_buf: Optional[FilePathOrBuffer[str]] = None,
+        sep: str = ",",
+        na_rep: str = "",
+        float_format: Optional[str] = None,
         cols=None,
-        header=True,
-        index=True,
-        index_label=None,
-        mode="w",
-        encoding=None,
-        compression="infer",
-        quoting=None,
+        header: Union[bool, Sequence[Hashable]] = True,
```

Comment:
question for @WillAyd - do we want ` Sequence[Optional[Hashable]]] ` or is this for readability and only add `Optional` when mypy complains?

Reply:
Personal preference for only when complaining but it isn't strong if you feel otherwise

Reply:
fine with that. (although may need to make exception for public apis)